src/parser_urdf_TEST.cc

```diff
@@ -994,6 +995,1096 @@ TEST(URDFParser, ParseWhitespace)
+        "have a fixed parent joint, unable to be converted into a frame in "
+        "sdf");
+    EXPECT_PRED2(sdf::testing::contains, buffer.str(),
+        "urdf2sdf: link[link1] has no <inertial> block defined, [1] children "
+        "links ignored");
+    EXPECT_PRED2(sdf::testing::contains, buffer.str(),
+        "urdf2sdf: link[link1] has no <inertial> block defined, [1] children "
+        "joints ignored");
+    EXPECT_PRED2(sdf::testing::contains, buffer.str(),
+        "urdf2sdf: link[link1] has no <inertial> block defined, not modeled in "
+        "sdf");
+
+    tinyxml2::XMLElement *sdf = sdfResult.FirstChildElement("sdf");
+    ASSERT_NE(nullptr, sdf);
+    tinyxml2::XMLElement *model = sdf->FirstChildElement("model");
+    ASSERT_NE(nullptr, model);
+    tinyxml2::XMLElement *frame = model->FirstChildElement("frame");
+    EXPECT_EQ(nullptr, frame);
+    tinyxml2::XMLElement *link = model->FirstChildElement("link");
+    EXPECT_EQ(nullptr, link);
+    tinyxml2::XMLElement *joint = model->FirstChildElement("joint");
+    EXPECT_EQ(nullptr, joint);
+  }
+}
+
+TEST(URDFParser, ConvertIntermediateLinkWithZeroMassToFrame)
+{
+  // Redirect sdfwarn output
+  std::stringstream buffer;
+  sdf::testing::RedirectConsoleStream redir(
+      sdf::Console::Instance()->GetMsgStream(), &buffer);
+#ifdef _WIN32
+  sdf::Console::Instance()->SetQuiet(false);
+  sdf::testing::ScopeExit revertSetQuiet(
+      []
+      {
+        sdf::Console::Instance()->SetQuiet(true);
+      });
+#endif
+
+  // intermediate link with no fixed joint
+  {
+    // clear the contents of the buffer
+    buffer.str("");
+
+    std::string str = R"(
+      <robot name='test_robot'>
+        <link name='link1'>
+          <inertial>
+            <mass value='0.1' />
+            <origin rpy='1.570796326794895 0 0' xyz='0.123456789123456 0 0.0' />
+            <inertia ixx='0.01' ixy='0' ixz='0' iyy='0.01' iyz='0' izz='0.01' />
+          </inertial>
+        </link>
+        <link name='link2'/>
+        <joint name='joint1_2' type='continuous'>
+          <parent link='link1' />
+          <child  link='link2' />
+          <origin xyz='0.0 0.0 0.0' rpy='0.0 0.0 1.57'/>
+        </joint>
+        <link name='link3'>
+          <inertial>
+            <mass value='0.1' />
+            <origin rpy='1.570796326794895 0 0' xyz='0.123456789123456 0 0.0' />
+            <inertia ixx='0.01' ixy='0' ixz='0' iyy='0.01' iyz='0' izz='0.01' />
+          </inertial>
+        </link>
+        <joint name='joint2_3' type='continuous'>
+          <parent link='link2' />
+          <child  link='link3' />
+          <origin xyz='0.0 0.0 0.0' rpy='0.0 0.0 1.57'/>
+        </joint>
+      </robot>)";
+
+    sdf::URDF2SDF parser;
+    sdf::ParserConfig defaultConfig;
+    tinyxml2::XMLDocument sdfResult;
+    parser.InitModelString(str, defaultConfig, &sdfResult);
+
+    // parent joint is not fixed, conversion to frame will fail
+    EXPECT_PRED2(sdf::testing::contains, buffer.str(),
```

Comment:
Should this test block be in the test above since it fails?

Reply:
I've re-wrote most the of the test to be more specific to the cases we discussed